server/app/lib/validations.js

```diff
@@ -22,4 +35,62 @@ export function validatePhone(phone) {
 
 export function validateEmail(email) {
   return EmailValidator.validate(email);
-}
\ No newline at end of file
+}
+
+export async function validateUniquePhone(modelName, phone, existingId = null) {
+  return validateUnique(modelName, { phone: normalizePhone(phone) }, existingId);
+}
+
+/**
+ * Returns true if a node with these properties does not yet exist,
+ * or if it only matches the given ID (useful when updating an existing node).
+ */
+export async function validateUnique(modelName, properties, existingId = null) {
+  if (!ENFORCE_UNIQUE) {
+    return true;
+  }
+  const all = await neode.all(modelName, properties);
+  if (all.length === 0) return true;
+  if (all.length > 1) return false;
+  const only = all[0];
+  return existingId == null || only.get('id') === existingId;
```

Comment:
Sometimes `only[0]` is either null, or doesn't have `.get()`. Not sure exactly what is happening.